WearTilesKotlin/app/src/main/java/com/example/wear/tiles/messaging/MessagingTileService.kt

```diff
@@ -190,39 +190,87 @@ class MessagingTileService : TileProviderService() {
-                .addContent(contactLayout(contacts[0], deviceParameters))
+                .addContent(
+                    contactLayout(
+                        contact = contacts[0],
+                        deviceParameters = deviceParameters,
+                        clickable = Clickable.builder()
+                            .setOnClick(ActionBuilders.LoadAction.builder())
+                            .build()
+                    )
+                )
                 .addContent(Spacer.builder().setWidth(SPACING_CONTACTS))
-                .addContent(contactLayout(contacts[1], deviceParameters))
+                .addContent(
+                    contactLayout(
+                        contact = contacts[1],
+                        deviceParameters = deviceParameters,
+                        clickable = Clickable.builder()
+                            .setOnClick(ActionBuilders.LoadAction.builder())
+                            .build()
+                    )
+                )
                 .addContent(Spacer.builder().setWidth(SPACING_CONTACTS))
-                .addContent(contactLayout(contacts[2], deviceParameters))
+                .addContent(
+                    contactLayout(
+                        contact = contacts[2],
+                        deviceParameters = deviceParameters,
+                        clickable = Clickable.builder()
+                            .setOnClick(ActionBuilders.LoadAction.builder())
+                            .build()
+                    )
+                )
         )
         .addContent(
             Row.builder()
-                .addContent(contactLayout(contacts[3], deviceParameters))
+                .addContent(
+                    contactLayout(
+                        contact = contacts[3],
+                        deviceParameters = deviceParameters,
+                        clickable = Clickable.builder()
+                            .setOnClick(ActionBuilders.LoadAction.builder())
+                            .build()
+                    )
+                )
                 .addContent(Spacer.builder().setWidth(SPACING_CONTACTS))
                 .addContent(searchLayout())
         )
+        .setModifiers(
+            Modifiers.builder()
+                .setSemantics(
+                    ModifiersBuilders.Semantics.builder()
+                        .setContentDescription(getString(R.string.tile_messaging_label))
+                )
+        )
         .build()
 
-    private fun contactLayout(contact: Contact, deviceParameters: DeviceParameters) = Box.builder().apply {
+    private fun contactLayout(
+        contact: Contact,
+        deviceParameters: DeviceParameters,
+        clickable: Clickable
+    ) = Box.builder().apply {
+        val modifiersBuilder = Modifiers.builder()
+            .setClickable(clickable)
+            .setSemantics(
+                ModifiersBuilders.Semantics.builder()
+                    .setContentDescription(contact.initials)
```

Comment:
Since this is a content description instead, should we use the full name rather than initials?

Reply:
Updated in the latest commit!